pkg/traceql/engine.go

```diff
@@ -61,59 +58,46 @@ func (e *Engine) Execute(ctx context.Context, searchReq *tempopb.SearchRequest,
 			break
 		}
 
-		span.LogKV("msg", "iterator.Next", "rootSpanName", spanSet.RootSpanName, "rootServiceName", spanSet.RootServiceName, "spans", len(spanSet.Spans))
-
-		spanSet = e.validateSpanSet(spanSetFilter, spanSet)
-		if spanSet == nil {
+		ss, err := rootExpr.Pipeline.evaluate([]Spanset{*spanSet})
+		if err != nil {
+			span.LogKV("msg", "pipeline.evaluate", "err", err)
```

Comment:
Wondering if it makes sense to fail the query on engine instead, to make failures more apparent.   In this log message let's go ahead and include some other help info like the pipeline itself (stringer will reprint the query).

Reply:
What do you mean with fail the query on engine instead? The query will fail here if data from the storage layer doesn't agree with the query (e.g. attribute has the wrong type), we can't handle these earlier.
I do agree we should improve the validation functions a bit and block some queries in the query-frontend.

Reply:
I mean to exit and return the error instead of logging and continuing.   Can you clarify a little what you mean the case where an attribute type doesn't agree with the query?  TraceQL is designed with no type casting, and only matches when the types are compatible:  i.e. ` { http.status_code >= 500 }` should simply ignore non-numeric values for `http.status_code` (BinaryOperation return false).   Would that generate an error?    The benefit of returning the error immediately is that it will be clearer to identify undefined or improperly defined engine logic.

Reply:
I was confusing this with the type check we do in the validation step. During validation `{ "foo"  = 5 }` will return an error, but when evaluating a span mismatching types will just return `false`. I'll change the logic to fail fast here: if evaluation errors, just return that to the caller. This way will be able to see issues faster.